client.go

```diff
@@ -220,3 +220,29 @@ func (fc *Client) LookupServerKeys(
 	}
 	return result, nil
 }
+
+// CreateMediaDownloadRequest creates a request for media on a homeserver and returns the http.Response or an error
+func (fc *Client) CreateMediaDownloadRequest(matrixServer ServerName, mediaID string) (*http.Response, error) {
+	dnsResult, err := LookupServer(matrixServer)
+	if err != nil {
+		if dnsErr, ok := err.(*net.DNSError); ok && dnsErr.Timeout() {
+			return nil, fmt.Errorf("DNS lookup for homeserver at %v timed out", matrixServer)
+		}
+		return nil, fmt.Errorf("error during server lookup: %q", err)
+	}
+
+	remoteReqAddr := "https://" + dnsResult.Addrs[0] + "/_matrix/media/v1/download/" + string(matrixServer) + "/" + mediaID
+	remoteReq, err := http.NewRequest("GET", remoteReqAddr, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	remoteReq.Header.Set("Host", string(matrixServer))
+
+	resp, err := fc.client.Do(remoteReq)
```

Comment:
The http client in `fc.client` has been modified to support a "matrix://" URL scheme that handles the server lookup so all of the above code can be replaced with 

```go
requestURL := "matrix://" + string(matrixServer) + "/_matrix/media/v1/download/" + string(matrixServer) + "/" + mediaID
resp, err := fc.client.GET(requestURL)
```